tensorflow_serving/util/json_tensor.cc

```diff
@@ -896,6 +927,13 @@ Status AddSingleValueAndAdvance(const TensorProto& tensor, bool string_as_bytes,
       success = writer->Int(tensor.int_val(*offset));
       break;
 
+    case DT_HALF:
+      int src = tensor.half_val(*offset);
+      float dst;
+      dst = toFloat16( &src, &dst);
```

Comment:
dont roll your own version for float16 conversion (its hard to get that correct). instead use standard @FP16 library that is used in TF. specifically use fp16_ieee_from_fp32_value() API (from fp16.h), and write the output as decimal number.

example code:
https://github.com/tensorflow/tensorflow/blob/2ba6502de549c20c7498f133792cf3223eabc274/tensorflow/lite/delegates/gpu/common/convert.cc#L303

you can refer to the fp16 library via @FP16 bazel target in the BUILD file.

also your change is incomplete. you need to handle input conversion (json -> tensor) for DT_HALF. please update AddValueToTensor() method.

and finally add unit tests for your code.

Reply:
Thank you , I'll try.